rust-runtime/aws-smithy-runtime/src/client/http/body/minimum_throughput/throughput.rs

```diff
@@ -97,90 +97,269 @@ impl From<(u64, Duration)> for Throughput {
     }
 }
 
-#[derive(Clone)]
+/// Represents a bin (or a cell) in a linear grid that represents a small chunk of time.
+#[derive(Copy, Clone, Debug)]
+enum Bin {
+    /// There is no data in this bin.
+    Empty,
+
+    /// No polling took place during this bin.
+    NoPolling,
+
+    /// This many bytes were transferred during this bin.
+    TransferredBytes(u64),
+
+    /// The user/remote was not providing/consuming data fast enough during this bin.
+    ///
+    /// The number is the number of bytes transferred, if this replaced TransferredBytes.
+    Pending(u64),
+}
```

Comment:
So a bin transitions from `TransferredBytes` into `Pending` if the last thing that happened in the bin was `Poll::Pending`?

Reply:
Yeah, `Pending` is the highest priority in the `Bin::merge` function below.